### PR TITLE
NP-581:Move functionality to its own package

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: gradlew [hunk 1]
```diff
@@ -1,5 +1,21 @@
 #!/usr/bin/env sh
 
+#
```
Comment (axthosarouris) [inline gradlew:3]: File is automatically generated by gradle. The difference is due to the upgrade to gralde 6.2.2

File: README.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
 # Crossref data model
 
-Java classes for deserializing Crossref JSON.
\ No newline at end of file
+Java classes for deserializing Crossref JSON.
+
+
+Details: https://docs.google.com/document/d/1PelmerlNDrMQqUvCTRwuEGsxQMtXQzNVab7ZyWlekqo/edit?usp=sharing
```
Comment (axthosarouris) [inline README.md:6]: Just a temporary link for the specs file

File: build.gradle [hunk 3]
```diff
@@ -5,15 +5,33 @@ plugins {
...
-    mavenCentral()
 }
 
-dependencies {
-    implementation group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.10.3'
-    implementation group: 'com.fasterxml.jackson.core', name: 'jackson-annotations', version: '2.10.3'
-    implementation group: 'com.fasterxml.jackson.core', name: 'jackson-core', version: '2.10.3'
-    testCompile group: 'junit', name: 'junit', version: '4.12'
-}
+subprojects {
```
Comment (axthosarouris) [inline build.gradle:14]: One module will be the functionality, the other the lambda function caller

File: data-model/src/test/java/no/unit/crossref/Integration.java [hunk 4]
```diff
@@ -3,7 +3,8 @@
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import no.unit.crossref.model.ApiResponse;
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
```
Comment (axthosarouris) [inline data-model/src/test/java/no/unit/crossref/Integration.java:6]: use of Junit 5

File: build.gradle [hunk 5]
```diff
@@ -5,15 +5,33 @@ plugins {
...
+apply plugin: 'java'
+
+    repositories {
+        jcenter()
+    }
+    dependencies {
+        implementation group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.10.3'
```
Comment (janbj) [inline build.gradle:21]: Use variable for common versions

**Review by janbj — APPROVED** [review]
Looks good

**Pull request merged**